sync_cleared_for_campus.R

```diff
@@ -0,0 +1,170 @@
+library(tidyverse)
+library(dotenv)
+library(REDCapR)
+library(lubridate)
+library(rlang)
+
+source("functions.R")
+
+# uncomment to read data from production
+#load_dot_env('prod.env')
+
+# get_case_records suitability is questionable
+case_records <- redcap_read_oneshot(redcap_uri = Sys.getenv("URI"),
+            token = Sys.getenv("CASE_TOKEN"),
+            guess_max = 3000)$data %>%
+    mutate(case_emp_ufid = as.character(case_emp_ufid)) # type change for dplyr
+
+# get_contact_records is not suitable
+contact_records <- redcap_read_oneshot(redcap_uri = Sys.getenv("URI"),
+                                               token = Sys.getenv("CONTACT_TOKEN"),
+                                               guess_max = 3000)$data
+
+# FIXME: not using oneshot results in type coercion error
+result_records <- redcap_read_oneshot(redcap_uri = Sys.getenv("URI"),
+        token = Sys.getenv("RESULT_TOKEN"),
+        guess_max = 3000)$data
```

Comment:
```suggestion
result_fields <- c("record_id", "redcap_repeat_instance", "redcap_repeat_instrument", "cleared_for_campus", "cleared_for_campus_timestamp")
result_records <- redcap_read_oneshot(redcap_uri = Sys.getenv("URI"),
        token = Sys.getenv("RESULT_TOKEN"),
        fields = result_fields,
        guess_max = 3000)$data %>%
    mutate(cleared_for_campus_timestamp = as.POSIXct.Date(cleared_for_campus_timestamp)) %>%
    mutate(cleared_for_campus = as.character(cleared_for_campus))
```

This will allow you to read completely NA data from results

Reply:
This is much faster than not passing any fields at all, I also noticed that while not passing fields I get _more records_ than actually exist in my local project.

`as.POSIXct.Date` make some strange assumptions, an adjustment is needed:  
![image](https://user-images.githubusercontent.com/20332546/84918707-f4f89d80-b08e-11ea-996b-c1fad155604f.png)

Reply:
@ChemiKyle, please take the parts of my suggestion and use them as you see fit. Discard anything you don't like.